File: adaptors/python/config/config_manager.py
import os
from typing import Dict, Any, Optional

class ConfigManager:
    _config: Optional[Dict[str, str]] = None
    _environment: str = 'development'
    
    @classmethod
    def find_env_file(cls) -> Optional[str]:
        current_dir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
        while current_dir != '/':
            jsondb_dir = os.path.join(current_dir, 'jsondb')
            if os.path.exists(jsondb_dir):
                env_file = os.path.join(jsondb_dir, f'.env.{cls._environment}')
                if os.path.exists(env_file):
                    return env_file
            current_dir = os.path.dirname(current_dir)
        return None

    @classmethod
    def load_config(cls) -> Dict[str, str]:
        env_file = cls.find_env_file()
        if not env_file:
            raise FileNotFoundError(f"Environment file not found for {cls._environment}")
            
        config = {}
        with open(env_file, 'r') as f:
            for line in f:
                line = line.strip()
                if not line or line.startswith('#'):
                    continue
                key, value = line.split('=', 1)
                config[key.strip()] = value.strip().strip("'").strip('"')
        return config

    @classmethod
    def get(cls, key: str, default: Any = None) -> Any:
        if cls._config is None:
            try:
                cls._config = cls.load_config()
            except Exception as e:
                print(f"Failed to load environment: {e}")
                return default
        return cls._config.get(key, default)
    
    @classmethod
    def all(cls) -> Dict[str, str]:
        if cls._config is None:
            try:
                cls._config = cls.load_config()
            except Exception as e:
                print(f"Failed to load environment: {e}")
                return {}
        return cls._config

File: adaptors/nodejs/config/ConfigManager.js
const { EnvLoader } = require('./EnvLoader');

class ConfigManager {
    static config = null;
    static environment = process.env.NODE_ENV || 'development';

    static initialize() {
        if (!this.config) {
            try {
                this.config = EnvLoader.load(this.environment);
            } catch (error) {
                console.error('Failed to load configuration:', error.message);
                this.config = {};
            }
        }
    }

    /**
     * Get a configuration value by key
     * @param {string} key - Configuration key
     * @param {*} defaultValue - Default value if key not found
     * @returns {*} Configuration value or default value
     */
    static get(key, defaultValue = null) {
        if (!this.config) {
            this.initialize();
        }
        return this.config[key] ?? defaultValue;
    }

    /**
     * Get all configuration values
     * @returns {Object} All configuration values
     */
    static all() {
        if (!this.config) {
            this.initialize();
        }
        return { ...this.config };
    }

    /**
     * Set a configuration value
     * @param {string} key - Configuration key
     * @param {*} value - Configuration value
     */
    static set(key, value) {
        if (!this.config) {
            this.initialize();
        }
        this.config[key] = value;
    }

    /**
     * Check if a configuration key exists
     * @param {string} key - Configuration key
     * @returns {boolean} True if key exists
     */
    static has(key) {
        if (!this.config) {
            this.initialize();
        }
        return key in this.config;
    }
}

module.exports = { ConfigManager };